### PR TITLE
Views

### PR DESCRIPTION
**Pull request by mpa-mxiang**
For this PR, I:

- Use preprocessed HTML files with embedded Ruby code.
- Use layouts and templates for shared content.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
     <% @users.each do |user| %>
-      <li>
+      <li class="user-item">
         <img src="<%= user.photo %>" alt="<%= user.name %>">
         <h1><%= user.name %></h1>
-        <p>Number of Posts: <%= user.posts.count %></p>
+        <p class="user-info">Number of Posts: <%= user.posts.count %></p>
       </li>
     <% end %>
   </ul>
```
Comment (julie-ify) [inline app/views/users/index.html.erb]: - [x] The users on the index page should have links that take the user to each user's show page. On the layout page, you can see there is space between the picture and the user details. And the number of posts part is towards the right of the container.

- [x] Kindly ensure your design follows [the layout](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_root_users.png?token=GHSAT0AAAAAACG3ADWRWSADJVFDEMCUEZDWZHM4BRQ) for the user index page

  image

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -2,10 +2,11 @@
 <html>
 <head>
   <title>User Details</title>
+  <link rel="stylesheet" type="text/css" href="/assets/users.css"> <!-- Include your CSS file -->
 </head>
 <body>
   <h1>User Details</h1>
-  <div>
+  <div class="user-details">
```
Comment (julie-ify) [inline app/views/users/show.html.erb]: - [x] Kindly note that the recent posts of the user should be displayed on the user show page, using the recent_posts method in the User model.

- [x] Kindly ensure the design of the user show page matches the show page layout design.

- [x] This page should also have a See All Posts button that takes the user to the user's posts index page.

  image

  image

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -2,15 +2,17 @@
...
+      <li class="post-item">
         <h2><%= post.title %></h2>
-        <p><%= post.created_at.strftime('%B %d, %Y') %></p>
-        <p><%= post.text %></p>
+        <p class="post-info">Author: <%= post.author.name %></p>
+        <p class="post-info">Created At: <%= post.created_at.strftime('%B %d, %Y') %></p>
+        <p class="post-text"><%= post.text %></p>
       </li>
     <% end %>
   </ul>
```
Comment (julie-ify) [inline app/views/posts/index.html.erb]: - [x] Kindly ensure that the posts index page is improved to fit the required design layout for it.

  image

  image

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -2,15 +2,17 @@
...
     <% @posts.each do |post| %>
-      <li>
+      <li class="post-item">
         <h2><%= post.title %></h2>
-        <p><%= post.created_at.strftime('%B %d, %Y') %></p>
-        <p><%= post.text %></p>
+        <p class="post-info">Author: <%= post.author.name %></p>
+        <p class="post-info">Created At: <%= post.created_at.strftime('%B %d, %Y') %></p>
+        <p class="post-text"><%= post.text %></p>
       </li>
```
Comment (julie-ify) [inline app/views/posts/index.html.erb]: - [x] The recent comments of a post should also be displayed on the posts index page, using the recent_comments method in the Post model.

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -2,15 +2,17 @@
...
+  <link rel="stylesheet" type="text/css" href="/assets/stylesheets/posts.css">
 </head>
 <body>
   <h1>All Posts</h1>
-  <ul>
+  <ul class="post-list">
     <% @posts.each do |post| %>
-      <li>
+      <li class="post-item">
         <h2><%= post.title %></h2>
```
Comment (julie-ify) [inline app/views/posts/index.html.erb]: - [x] The post title/body on the posts index page should be a link to the post's show page

File: app/views/posts/show.html.erb [hunk 6]
```diff
@@ -2,12 +2,15 @@
...
-  <p>Created At: <%= @post.created_at.strftime('%B %d, %Y') %></p>
-  <p><%= @post.text %></p>
+  <div class="post-details">
+    <h2><%= @post.title %></h2>
+    <p class="post-info">Author: <%= @post.author.name %></p>
+    <p class="post-info">Created At: <%= @post.created_at.strftime('%B %d, %Y') %></p>
+    <p class="post-text"><%= @post.text %></p>
+  </div>
 </body>
 </html>
```
Comment (julie-ify) [inline app/views/posts/show.html.erb]: - [x] Kindly improve the post show page to follow the required layout.

  image

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @mpa-mxiang,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 1 above)
Comment (mpa-mxiang) [reply app/views/users/index.html.erb]: Hi @julie-ify , it shows me `404: Not Found` when I open the link, could you please update it?

**Review by Davidosky007 — APPROVED** [review]
Hi @mpa-mxiang 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All view pages works 👍 
* Well written PR 👍 
* No linter errors 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mpa-mxiang** [comment]
Thank you @Davidosky007 for the approval!

**Pull request merged**